### PR TITLE
Remove inline JavaScript from templates

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Moves GTM snippet to an external JS file
- Removes `platform_image.html` which no longer seems to be used (no idea as to it's origin, hasn't been touched in 3 years).
## Bugzilla link

https://bugzilla.mozilla.org/show_bug.cgi?id=1307567
## Testing
- I need to put this up on demo and confirm all the GA stuff seems to be working still.
## Checklist
- [ ] Requires l10n changes.
- [ ] Related functional & integration tests passing.

### CODE REVIEW

**Comment by alexgibson** [comment]
Note there's [one other piece of inline JS](https://github.com/mozilla/bedrock/blob/master/bedrock/newsletter/templates/newsletter/hacks.mozilla.org.html#L79-L82) that I already took care of in #4406

File: bedrock/base/templates/includes/google-analytics.html [hunk 1]
```diff
@@ -5,16 +5,6 @@
...
-// @see https://bugzilla.mozilla.org/show_bug.cgi?id=1217896 for more details
-if (!_dntEnabled()) {
-  (function(w,d,s,l,i,j,f,dl,k,q){
-    w[l]=w[l]||[];w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'});f=d.getElementsByTagName(s)[0];
-    k=i.length;q='//www.googletagmanager.com/gtm.js?id=@&l='+(l||'dataLayer');
-    while(k--){j=d.createElement(s);j.async=!0;j.src=q.replace('@',i[k]);f.parentNode.insertBefore(j,f);}
-  }(window,document,'script','dataLayer',['{{settings.GTM_CONTAINER_ID}}']));
-}
-</script>
+  <script type="text/javascript" src="{{ static('js/base/gtm-snippet.js') }}"></script>
```
Comment (alexgibson) [inline bedrock/base/templates/includes/google-analytics.html]: I think this is a good suggestion (as long as minifying the already minified snippet code doesn't cause any issues)

**Comment by alexgibson** [comment]
I pushed this to demo4: https://www-demo4.allizom.org/en-US/

Will ping peter in the bug to see if he can find some time to test this to make sure it doesn't break anything. As far as requests to it all appears to be fine.

**Comment by alexgibson** [comment]
We have [confirmation from Peter](https://bugzilla.mozilla.org/show_bug.cgi?id=1307567#c5) that GA events are working as expected on the demo branch. Removing the DNM label.

**Comment by jpetto** [comment]
One small question, but otherwise looks good. r+wpc (with possible corrections 😉)

File: media/js/base/gtm-snippet.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global _dntEnabled */
+
+(function() {
+    var GTM_CONTAINER_ID = document.getElementsByTagName('html')[0].getAttribute('data-gtm-container-id');
+
+    // If doNotTrack is not enabled, it is ok to add GTM
+    // @see https://bugzilla.mozilla.org/show_bug.cgi?id=1217896 for more details
+    if (!_dntEnabled() && GTM_CONTAINER_ID) {
```
Comment (alexgibson) [inline media/js/base/gtm-snippet.js]: Good call 👻

**Comment by alexgibson** [comment]
@jpetto updated thanks

**Comment by jpetto** [comment]
More spooky r+'s. 💀

**Pull request merged**